### PR TITLE
fix(ripple): default to using background-color: currentColor for before/after ripple

### PR DESCRIPTION
**Pull request by moog16**
fixes https://github.com/material-components/material-components-web/issues/4571

### CODE REVIEW

**Comment by moog16** [comment]
The ripple color in button should be the same as the text color as shown in this glitch: https://glitch.com/edit/#!/various-ocelot?path=index.html:9:0. In the first example if we change the CSS var, --mdc-theme-primary, it updates both text and ripple color. In the second example, If we just change the ink color it doesn't update the ripple color.

**Comment by moog16** [comment]
@abhiomkar while fixing #4527 I'm finding that I need to customize the ripple. If I have background: currentColor, I wouldn't have to customize the ripple color. Can we pull this one out of the icebox?

File: packages/mdc-ripple/_mixins.scss [hunk 1]
```diff
@@ -184,18 +184,22 @@
...
+        } @else {
+          // If a color with 0 alpha is specified, don't render the ripple pseudo-elements at all.
+          // This avoids unnecessary transitions and overflow.
+          content: none;
+        }
       } @else {
-        // If a color with 0 alpha is specified, don't render the ripple pseudo-elements at all.
-        // This avoids unnecessary transitions and overflow.
-        content: none;
+        background-color: currentColor;
```
Comment (abhiomkar) [inline packages/mdc-ripple/_mixins.scss]: Seems like unsafe assumption to use inherited color for ripple where the inherited color is unknown.

Comment (moog16) [reply packages/mdc-ripple/_mixins.scss]: In the majority of our components the background-color is currentColor. We will provide the mixin to override this. Also this IFF the $color argument is not provided, which was not an option before. Meaning this is backwards compatible.

**Comment by moog16** [comment]
@abhiomkar confirmed that on the icon-button, if we change the wrapping element's color to a different color with `!important`, this does not cascade down to the element. Example from screenshot tests:
```
<div style='color: orange !important'>
  <button class="mdc-icon-button mdc-icon-button--on mdc-theme--primary" aria-label="Add to favorites" aria-pressed="true">
    :before
    <i class="material-icons mdc-icon-button__icon mdc-icon-button__icon--on">favorite</i>
    <i class="material-icons mdc-icon-button__icon">favorite_border</i>
  </button>
</div>
```
The :before element's background-color is still button's color via the `currentColor` prop.

**Comment by moog16** [comment]
Closing - stale PR

**Pull request closed without merging**